parachain/primitives/runtime-common/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+//! # Runtime Common
+//!
+//! Common traits and types shared by runtimes.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use bp_rococo::AccountId;
+use core::marker::PhantomData;
+use frame_support::traits::Get;
+use snowbridge_core::{outbound::SendMessageFeeProvider, sibling_sovereign_account_raw};
+use xcm::prelude::*;
+use xcm_builder::{deposit_or_burn_fee, HandleFee};
+use xcm_executor::traits::{FeeReason, TransactAsset};
+
+/// A `HandleFee` implementation that takes fees from `ExportMessage` XCM instructions
+/// to Snowbridge and splits off the local remote fee and deposits it to the origin
+/// parachain sovereign account. The local fee is then returned back to be handled by
+/// the next fee handler in the chain. Most likely the treasury account.
+pub struct XcmExportFeeToSnowbridge<TokenLocation, EthereumNetwork, AssetTransactor, OutboundQueue>(
+	PhantomData<(TokenLocation, EthereumNetwork, AssetTransactor, OutboundQueue)>,
+);
+
+impl<
+		TokenLocation: Get<MultiLocation>,
+		EthereumNetwork: Get<NetworkId>,
+		AssetTransactor: TransactAsset,
+		OutboundQueue: SendMessageFeeProvider<Balance = bp_rococo::Balance>,
+	> HandleFee
+	for XcmExportFeeToSnowbridge<TokenLocation, EthereumNetwork, AssetTransactor, OutboundQueue>
+{
+	fn handle_fee(
+		fees: MultiAssets,
+		context: Option<&XcmContext>,
+		reason: FeeReason,
+	) -> MultiAssets {
+		let token_location = TokenLocation::get();
+
+		// Check the reason to see if this export is for snowbridge.
+		let snowbridge_export = matches!(
+			reason,
+			FeeReason::Export { network: bridged_network, destination }
+				if bridged_network == EthereumNetwork::get() && destination == Here
+		);
+
+		// Get the parachain sovereign from the `context`.
+		let maybe_para_sovereign = if let Some(XcmContext {
```

Comment:
if `XcmContext` is `None`, we should perhaps exit early, and allow the next fee handler to process the fee. This may make the code look a bit cleaner.

Reply:
Addressed in https://github.com/Snowfork/snowbridge/pull/999/commits/e0d356f1c360b7f4ff8b4a2ffe055dc326d9d612